go/registry/api/runtime.go

```diff
@@ -316,6 +341,61 @@ type Runtime struct { // nolint: maligned
 
 	// Staking stores the runtime's staking-related parameters.
 	Staking RuntimeStakingParameters `json:"staking,omitempty"`
+
+	// GovernanceModel specifies the runtime governance model.
+	GovernanceModel RuntimeGovernanceModel `json:"governance_model"`
+}
+
+// RuntimeGovernanceModel specifies the runtime governance model.
+type RuntimeGovernanceModel uint8
+
+const (
+	GovernanceInvalid   RuntimeGovernanceModel = 0
+	GovernanceEntity    RuntimeGovernanceModel = 1
+	GovernanceRuntime   RuntimeGovernanceModel = 2
+	GovernanceConsensus RuntimeGovernanceModel = 3
+
+	GovernanceMax = GovernanceConsensus
+
+	gmInvalid   = "invalid"
+	gmEntity    = "entity"
+	gmRuntime   = "runtime"
+	gmConsensus = "consensus"
+)
+
+// String returns a string representation of a runtime governance model.
+func (gm RuntimeGovernanceModel) String() string {
+	switch gm {
+	case GovernanceInvalid:
+		return gmInvalid
+	case GovernanceEntity:
+		return gmEntity
+	case GovernanceRuntime:
+		return gmRuntime
+	case GovernanceConsensus:
+		return gmConsensus
+	default:
+		return "[unsupported runtime governance model]"
+	}
+}
+
+func (gm RuntimeGovernanceModel) MarshalText() ([]byte, error) {
+	return []byte(gm.String()), nil
```

Comment:
This shouldn't just use `.String()` as that can in theory return the `[unsupported runtime governance model]`. It should instead return an error in this case.